From 4bdb0e604cdf28aa67b6336ba3ad0edac51e03b1 Mon Sep 17 00:00:00 2001
From: Ruslan Bay <67730802+ruslanbay@users.noreply.github.com>
Date: Tue, 17 Feb 2026 12:47:09 +0000
Subject: [PATCH 47/54] media: intel/ipu4: Count video devices for nr_queues
 initialization

Count all video device entities during graph walk in ipu_isys_video_prepare_streaming()
and increment nr_queues accordingly. Previously, nr_queues was only incremented in
isa_link_validate(), causing direct CSI-2 capture pipelines to have nr_queues=0,
which made the stream start hang indefinitely.

---
 drivers/media/pci/intel/ipu-isys-video.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys-video.c b/drivers/media/pci/intel/ipu-isys-video.c
index 1a6aeef7506a..af8325f00842 100644
--- a/drivers/media/pci/intel/ipu-isys-video.c
+++ b/drivers/media/pci/intel/ipu-isys-video.c
@@ -1898,11 +1898,14 @@ int ipu_isys_video_prepare_streaming(struct ipu_isys_video *av,
 	if (rval)
 		goto out_pipeline_stop;
 
-	/* Gather all entities in the graph. */
+	/* Gather all entities in the graph and count video queues. */
 	mutex_lock(&mdev->graph_mutex);
 	media_graph_walk_start(&graph, &av->vdev.entity);
-	while ((entity = media_graph_walk_next(&graph)))
+	while ((entity = media_graph_walk_next(&graph))) {
 		media_entity_enum_set(&ip->entity_enum, entity);
+		if (is_media_entity_v4l2_video_device(entity))
+			ip->nr_queues++;
+	}
 
 	mutex_unlock(&mdev->graph_mutex);
 
-- 
2.51.0

